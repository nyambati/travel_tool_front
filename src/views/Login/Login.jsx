import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { setCurrentUser } from '../../redux/actionCreator';
import andelaLogo from '../../images/andela-logo.svg';
import cover from '../../images/cover.svg';
import symbolG from '../../images/Google-white.svg';
import videoSymbol from '../../images/video.svg';
import fileSymbol from '../../images/file.svg';
import './Login.scss';
import TextLink from '../../components/text-link/TextLink';
import { authenticationMessage } from '../../helper/toast';

export class Login extends Component {

  componentDidMount() {
    this.authenticated();
  }

  authenticated = () => {
    const { isAuthenticated, history, setCurrentUser } = this.props;
    if(isAuthenticated){
     history.push('/request');
     authenticationMessage();
    }
    setCurrentUser();
  }

  login = () => {
    const url = `${process.env.REACT_APP_ANDELA_AUTH_HOST}/login?redirect_url=${process.env.REACT_APP_AUTH_REDIRECT_URL}`;
    window.location.replace(url);
  }

  render() {
    return (
      <div className="mdl-layout mdl-js-layout login-page">
        <div className="mdl-layout__content">
          <div className="mdl-grid mdl-grid--no-spacing">

            {/* Add title and login button on the login page */}
            <div className="mdl-cell mdl-cell--5-col">

              <img src={andelaLogo} alt="Andela Logo" className="login-page__andela-logo" />

              <p className="login-page__travel-request-text">
                  Travel Requests Made Easier
              </p>

              <button type="button" onClick={this.login} className="mdl-button mdl-js-button mdl-button--raised mdl-button--colored login-page__login-btn">
                <img src={symbolG} alt="Google Symbol" className="login-page__google-white" />
                <span className="login-page__login-to-get-started-text">
                 Login to Get Started
                </span>
              </button>

              {/* Add text link on the login page */}

              <TextLink
                imageSrc={videoSymbol}
                textLinkClass="login-page__how-to-book-a-trip-link"
                textClass="login-page__how-to-book-a-trip-text"
                altText="Video Symbol"
                text="How to book a trip"
              />
              
              <TextLink
                imageSrc={fileSymbol}
                textLinkClass="login-page__andela-travel-policy-link"
                textClass="login-page__andela-travel-policy"
                altText="File Symbol"
                text="Andela travel policy"
              />

            </div>

            {/* Add a road map on the login page */}
            <div className="mdl-cell mdl-cell--7-col">
              <img src={cover} alt="Road map" className="login-page__landing-page-map" />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  history: PropTypes.shape({}).isRequired,
  setCurrentUser: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    isAuthenticated: state.auth.isAuthenticated
  };
};

export default connect(mapStateToProps, { setCurrentUser })(Login);
