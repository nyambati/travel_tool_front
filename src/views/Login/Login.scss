.login-page {
  &__andela-logo {
    padding: 80px;
  }
  

  &__landing-page-map {
  width: 108%;
  }
  

  &__travel-request-text {
    padding: 59px 80px 55px 80px;
    height: 102px;
    width: 292px;
    color: #3359DB;
    font-family: "DIN Pro";
    font-size: 40px;
    font-weight: 600;
    line-height: 51px;
    font-style: normal;
    margin-bottom: 0px;
  }
  

  &__login-btn {
    padding: 0%;
    height: 50px;
    width: 252px;
    border-radius: 4px;
    background-color: #3359DB !important;
    margin-left: 80px;
    text-transform: none;
  }
  

  &__login-to-get-started-text {
    height: 41px;
    width: 162px;
    color: #FFFFFF;
    font-family: "DIN Pro";
    font-size: 18px;
    font-weight: 600;
    line-height: 40px;
  }
  

  &__how-to-book-a-trip-text {
    height: 21px;
    width: 129px;
    color: #3359DB;
    font-family: "DIN Pro";
    font-size: 16px;
    line-height: 21px;
    margin-left: 12.12px;
  }
  

  &__andela-travel-policy {
    height: 21px;
    width: 139px;
    color: #3359DB;
    font-family: "DIN Pro";
    font-size: 16px;
    line-height: 21px;
    margin-left: 12.12px;
  }
  

  &__how-to-book-a-trip-link {
    padding: 215px 80px 0px 80px;
    margin-bottom: 10px;
  }
  

  &__andela-travel-policy-link {
    padding: 0px 90px 0px 80px;
    margin-bottom: 0px;
  }


  &__google-white {
    margin: -7px 19px 0px 0px;
  }
}
