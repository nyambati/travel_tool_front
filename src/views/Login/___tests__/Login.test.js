import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../Login';

it('should render Login page correctly', () => {
const tree = renderer.create(<Login />).toJSON();
expect(tree).toMatchSnapshot();
});