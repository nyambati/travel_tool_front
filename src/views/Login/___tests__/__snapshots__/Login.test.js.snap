// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login page correctly 1`] = `
<div
  className="mdl-layout mdl-js-layout login-page"
>
  <div
    className="mdl-layout__content"
  >
    <div
      className="mdl-grid mdl-grid--no-spacing"
    >
      <div
        className="mdl-cell mdl-cell--5-col"
      >
        <img
          alt="Andela Logo"
          className="login-page__andela-logo"
          src="andela-logo.svg"
        />
        <p
          className="login-page__travel-request-text"
        >
          Travel Requests Made Easier
        </p>
        <button
          className="mdl-button mdl-js-button mdl-button--raised mdl-button--colored login-page__login-btn"
          type="button"
        >
          <img
            alt="Google Symbol"
            className="login-page__google-white"
            src="Google-white.svg"
          />
          <span
            className="login-page__login-to-get-started-text"
          >
            Login to Get Started
          </span>
        </button>
        <p
          className="login-page__how-to-book-a-trip-link"
        >
          <img
            alt="Video Symbol"
            src="video.svg"
          />
          <a
            className="login-page__how-to-book-a-trip-text"
            href="/"
          >
            How to book a trip
          </a>
        </p>
        <p
          className="login-page__andela-travel-policy-link"
        >
          <img
            alt="File Symbol"
            src="file.svg"
          />
          <a
            className="login-page__andela-travel-policy"
            href="/"
          >
            Andela travel policy
          </a>
        </p>
      </div>
      <div
        className="mdl-cell mdl-cell--7-col"
      >
        <img
          alt="Road map"
          className="login-page__landing-page-map"
          src="cover.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
